activerecord/test/cases/relation/field_ordered_values_test.rb

```diff
@@ -57,4 +57,12 @@ def test_in_order_of_after_regular_order
 
     assert_equal(order, posts.map(&:id))
   end
+
+  def test_in_order_of_with_out_of_bound_integer
+    max_integer = Post.type_for_attribute(:id).send(:max_value)
+    order = [max_integer, 1]
+    posts = Post.where(type: "Post").order(:type).in_order_of(:id, order)
```

Comment:
I would go with this more visual example taken from your issue report:

```suggestion
    posts = Post.where(type: "Post").order(:type).in_order_of(:id, [1, 9999999999999])
```

Reply:
Do you mind if I keep it as it is?

`9999999999999` from the example is now not big enough since the limit changed from 4 to 8 bytes

With dynamic out-of-bound integer, it can catchup whenever the upper bound changed

```ruby
FieldOrderedValuesTest#test_in_order_of_with_out_of_bound_integer:
ActiveModel::RangeError: 9223372036854775808 is out of range for ActiveRecord::ConnectionAdapters::SQLite3Adapter::SQLite3Integer with limit 8 bytes
```

Reply:
Makes sense. I don't know if the 64 bit integer limit will increase any time soon so you could hardcode `99_999_999_999_999_999_999` but if that feels like too long then feel free to keep as is and mark this comment as resolved.